src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -77,41 +91,295 @@
     @In
     private NUIManager nuiManager;
 
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private PrefabManager prefabManager;
+
+    @In
+    private DelayManager delayManager;
+
     private static final Logger logger = LoggerFactory.getLogger(BreedingSystem.class);
     private Random random = new FastRandom();
 
     /**
-     * Receives the BreedGooeyEvent when the "activated" gooey is chosen for breeding.
+     * This method is called when a gooey steps on a breeding block
+     *
+     * @param event     The OnEnterBlockEvent
+     * @param entity    the gooey entity
+     */
+    @ReceiveEvent
+    public void onEnterBlock(OnEnterBlockEvent event, EntityRef entity, GooeyComponent gooeyComponent) {
+        LocationComponent loc = entity.getComponent(LocationComponent.class);
+        Vector3f pos = loc.getWorldPosition();
+        pos.setY(pos.getY() -1);
+
+        EntityRef blockEntity = blockEntityRegistry.getExistingBlockEntityAt(new Vector3i(pos, RoundingMode.HALF_UP));
+
+        if (blockEntity.hasComponent(BreedingBlockComponent.class)) {
+            FollowComponent followComponent = entity.getComponent(FollowComponent.class);
+            BreedingBlockComponent breedingBlockComponent = blockEntity.getComponent(BreedingBlockComponent.class);
+
+            if (followComponent.entityToFollow != EntityRef.NULL && breedingBlockComponent.parentGooey == EntityRef.NULL && gooeyComponent.isCaptured) {
```

Comment:
The null checks can be simplified to `followComponent.entityToFollow.exists()`, likewise so can the breeding block entity check (negating of course)